src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/CSharpVirtualDocument.cs

```diff
@@ -84,5 +88,17 @@ public override VirtualDocumentSnapshot Update(IReadOnlyList<TextChange> changes
         }
 
         private CSharpVirtualDocumentSnapshot UpdateSnapshot() => new CSharpVirtualDocumentSnapshot(Uri, TextBuffer.CurrentSnapshot, HostDocumentSyncVersion);
+
+        private void MakeEmptyEdit()
+        {
+            var bufferLength = TextBuffer.CurrentSnapshot.Length;
+            using var edit = TextBuffer.CreateEdit(EditOptions.None, reiteratedVersionNumber: null, InviolableEditTag.Instance);
+            edit.Insert(bufferLength, " ");
+            edit.Apply();
+
+            using var revertEdit = TextBuffer.CreateEdit(EditOptions.None, reiteratedVersionNumber: null, InviolableEditTag.Instance);
+            revertEdit.Delete(bufferLength, 1);
+            revertEdit.Apply();
+        }
```

Comment:
I evaluated this vs appending the version at the end of the file and decided this was the least evil. Because,
- This only happens when there are 0 real changes vs when appending versions we'll probably do it for every update.
- This is the least destructive because we don't really add anything extra and we are guaranteed that we won't face weird issues when it comes to formatting etc.